sdks/python/apache_beam/io/parquetio_test.py

```diff
@@ -0,0 +1,376 @@
+from apache_beam.testing.util import assert_that
+from apache_beam.testing.util import equal_to
+from apache_beam.transforms.display import DisplayData
+from apache_beam.transforms.display_test import DisplayDataItemMatcher
+
+
+class TestParquet(unittest.TestCase):
+  _temp_files = []
+
+  @classmethod
+  def setUpClass(cls):
+    # Method has been renamed in Python 3
+    if sys.version_info[0] < 3:
+      cls.assertCountEqual = cls.assertItemsEqual
+
+  def setUp(self):
+    # Reducing the size of thread pools. Without this test execution may fail in
+    # environments with limited amount of resources.
+    filebasedsource.MAX_NUM_THREADS_FOR_SIZE_ESTIMATION = 2
+
+  def tearDown(self):
+    for path in self._temp_files:
+      if os.path.exists(path):
+        os.remove(path)
+        parent = os.path.dirname(path)
+        if not os.listdir(parent):
+          os.rmdir(parent)
+    self._temp_files = []
+
+  RECORDS = [{'name': 'Thomas',
+              'favorite_number': 1,
+              'favorite_color': 'blue'}, {'name': 'Henry',
+                                          'favorite_number': 3,
+                                          'favorite_color': 'green'},
+             {'name': 'Toby',
+              'favorite_number': 7,
+              'favorite_color': 'brown'}, {'name': 'Gordon',
+                                           'favorite_number': 4,
+                                           'favorite_color': 'blue'},
+             {'name': 'Emily',
+              'favorite_number': -1,
+              'favorite_color': 'Red'}, {'name': 'Percy',
+                                         'favorite_number': 6,
+                                         'favorite_color': 'Green'}]
+
+  SCHEMA = pa.schema([
+      ('name', pa.binary()),
+      ('favorite_number', pa.int64()),
+      ('favorite_color', pa.binary())
+  ])
+
+  def _record_to_columns(self, records, schema):
+    col_list = []
+    for n in schema.names:
+      column = []
+      for r in records:
+        column.append(r[n])
+      col_list.append(column)
+    return col_list
+
+  def _write_data(self,
+                  directory=None,
+                  prefix=tempfile.template,
+                  row_group_size=1000,
+                  codec='none',
+                  count=len(RECORDS)):
+
+    with tempfile.NamedTemporaryFile(
+        delete=False, dir=directory, prefix=prefix) as f:
+      len_records = len(self.RECORDS)
+      data = []
+      for i in range(count):
+        data.append(self.RECORDS[i % len_records])
+      col_data = self._record_to_columns(data, self.SCHEMA)
+      col_array = [pa.array(c) for c in col_data]
+      table = pa.Table.from_arrays(col_array, self.SCHEMA.names)
+      pq.write_table(table, f, row_group_size=row_group_size, compression=codec)
+
+      self._temp_files.append(f.name)
+      return f.name
+
+  def _write_pattern(self, num_files):
+    assert num_files > 0
+    temp_dir = tempfile.mkdtemp()
```

Comment:
You should be able to simplify by using the Beam test_utils module here. See https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/vcfio_test.py#L109.